components/chart/header/tagsInput.tsx

```diff
@@ -161,7 +187,29 @@ const TagsInput = () => {
         onRemove={onRemove}
         placeholder="Select tags"
       />
-      <button className='login' onClick={handleLogin}>login</button>
+      {typeof window !== 'undefined' && auth?.isAuthenticated() ? (
+        <>
+          <Switch
+            className="switch"
+            onChange={handleEditModeChange}
+            checked={editMode}
+            onColor="#699041"
+            offColor="#ccc"
+            onHandleColor="#fff"
+            offHandleColor="#fff"
+            handleDiameter={20}
+            uncheckedIcon={false}
+            checkedIcon={false}
+            boxShadow="0px 1px 5px rgba(0, 0, 0, 0.6)"
+            activeBoxShadow="0px 0px 1px 3px rgba(0, 0, 0, 0.2)"
+            height={20}
+            width={40}
+          />
+          <button className='login' onClick={handleUserLogOut}>logOut</button>
+        </>
+      ) : (
+        <button className='login' onClick={handleLogin}>login</button>
+      )}
```

Comment:
I think login/logout and edit mode should not be inside TagsInput because they are something else, let's add them to the header directly